Extensions/dnSpy.StringSearcher/StringReferencesService.cs

```diff
@@ -0,0 +1,172 @@
+	public class StringReferencesService : IStringReferencesService {
+		private readonly IDecompilerService decompilerService;
+		private readonly ITextElementProvider textElementProvider;
+		private readonly IClassificationFormatMapService classificationFormatMapService;
+		private readonly IDocumentTabService documentTabService;
+		private readonly Action<IEnumerable<StringReference>> addItems;
+		private readonly StringsControlVM vm;
+		private readonly Dispatcher dispatcher;
+		private ModuleDef[] selectedModules = [];
+
+		public StringsControl UIObject { get; }
+
+		object? IUIObjectProvider.UIObject => UIObject;
+
+		public IInputElement? FocusedElement => null;
+
+		public FrameworkElement? ZoomElement => UIObject.ListView;
+
+		public StringReference? CurrentReference => vm.SelectedStringReference;
+
+		[ImportingConstructor]
+		public StringReferencesService(
+			IDecompilerService decompilerService,
+			ITextElementProvider textElementProvider,
+			IClassificationFormatMapService classificationFormatMapService,
+			IDocumentTabService documentTabService,
+			IMenuService menuService,
+			IWpfCommandService wpfCommandService) {
+
+			this.decompilerService = decompilerService;
+			this.textElementProvider = textElementProvider;
+			this.classificationFormatMapService = classificationFormatMapService;
+			this.documentTabService = documentTabService;
+
+			UIObject = new StringsControl {
+				DataContext = vm = new StringsControlVM(this)
+			};
+
+			dispatcher = Dispatcher.CurrentDispatcher;
+
+			addItems = items => {
+				foreach (var item in items) {
+					vm.StringLiterals.Add(item);
+				}
+			};
+
+			menuService.InitializeContextMenu(
+				UIObject.ListView,
+				new Guid(StringSearcherConstants.GUID_STRINGS_LISTBOX),
+				new GuidObjectsProvider()
+			);
+
+			wpfCommandService.Add(new Guid(StringSearcherConstants.GUID_STRINGS_LISTBOX), UIObject.ListView);
+
+			UIObject.ListView.MouseDoubleClick += (_, _) => {
+				FollowSelectedReference((Keyboard.Modifiers & ModifierKeys.Control) != 0);
+			};
+			UIObject.ListView.KeyUp += (_, e) => {
+				if (e.Key == Key.Enter) {
+					FollowSelectedReference((Keyboard.Modifiers & ModifierKeys.Control) != 0);
+				}
+			};
+		}
+
+		private void FollowSelectedReference(bool newTab) {
+			if (vm.SelectedStringReference is { } selected) {
+				FollowReference(selected, newTab);
+			}
+		}
+
+		public void Analyze(IEnumerable<ModuleDef> modules) {
+			selectedModules = modules.ToArray();
+			AnalyzeSelectedModules();
+		}
+
+		public void Analyze(ModuleDef module) => Analyze([module]);
+
+		private void AnalyzeSelectedModules() {
+			var context = new StringReferenceContext(
+				decompilerService.Decompiler,
+				textElementProvider,
+				classificationFormatMapService.GetClassificationFormatMap("UIMisc") // TODO: replace string with AppearanceCategoryConstants.UIMisc
+			);
+
+			vm.StringLiterals.Clear();
+
+			Parallel.ForEach(selectedModules.SelectMany(x => x.GetTypes()), type => {
```

Comment:
The `AnalyzeSelectedModules` is executed on the UI thread which results in the enumerable that is passed into `Parallel.ForEach` to be enumerated on the UI thread as well. In the case of a assembly with 40000 types I tested with, this resulted in a significant 1-2 second UI freeze.

We should make sure to evaluate this enumerable on a seperate thread.